api/users.rb

```diff
@@ -77,3 +77,14 @@
   user.mark_as_read(source)
   user.reload.to_hash.to_json
 end
+
+post "#{APIPREFIX}/users/:user_id/retire" do |user_id|
+  return {}.to_json if not params["retired_username"]
+  user = User.find_by!(external_id: user_id)
+  user.update_attribute(:email, "")
+  user.update_attribute(:notification_ids, [])
+  user.update_attribute(:read_states, [])
+  user.unsubscribe_all
+  user.retire_all_comments(params["retired_username"])
+  user.update_attribute(:username, params["retired_username"])
```

Comment:
Yes - since user_id ("id" from LMS's auth_user table) is passed in, all these steps will be idempotent. The user_id will not change in the documents in the forum's "users" collection.